### PR TITLE
Monkey patch `Hyperparameter.sample_value()` to support quantization

### LINKED ISSUE #803: Implement quantization for ConfigSpace >= 1.0
ConfigSpace 1.0 hyperparameters do not have the quantization parameter `q=` anymore. In #802 we temporarily remove support for quantization in ConfigSpace. We need to add it back using the new `quantize()` family of functions of ConfigSpace.

### PR DESCRIPTION
**Pull request by motus**
Experimental implementation of quantization of tunable parameters

Closes #803

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py [hunk 1]
```diff
@@ -62,6 +62,7 @@
                 "type": "int",
                 "default": 2000000,
                 "range": [0, 1000000000],
+                "quantization": 1000000,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py:65]: Probably need tests to check this now, right?

Comment (motus) [reply mlos_bench/mlos_bench/tests/tunable_groups_fixtures.py:65]: absolutely. I'll add a ton more unit tests to this PR shortly

File: mlos_bench/mlos_bench/optimizers/convert_configspace.py [hunk 2]
```diff
@@ -149,6 +148,19 @@ def _tunable_to_configspace(
     else:
         raise TypeError(f"Invalid Parameter Type: {tunable.type}")
 
+    if tunable.quantization:
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/optimizers/convert_configspace.py:151]:
```suggestion
    if tunable.quantization:
        # Temporary workaround to dropped quantization support in ConfigSpace 1.0
        # See Also: https://github.com/automl/ConfigSpace/issues/390
```

**Review by bpkroth — CHANGES_REQUESTED** [review]
Waiting on tests and then let's merge this.

**Comment by motus** [comment]
superseded by #835

**Pull request closed without merging**